packages/strategies/stress.go

```diff
@@ -70,65 +87,220 @@ func (s *StressTest) modifyTransactions(input *types.Block, transactions types.T
+	}
+
+	val, ok := new(big.Int).SetString("100000000000000000000", 10)
+	if !ok {
+		panic("big.Int SetString failed")
+	}
+
+	dep.SourceHash = source.SourceHash()
+	dep.From = addr
+	dep.To = &addr
+	dep.Mint = val
+	dep.Value = val
+	dep.Gas = 21_000
+	dep.IsSystemTransaction = false
+
+	nonceMu.Lock()
+	nonces[i] += 1
+	nonceMu.Unlock()
+
+	return types.NewTx(&dep)
+}
+
+func (s *StressTest) randAddrIndex() (from *big.Int, to *big.Int, err error) {
+	for {
+		from, err = rand.Int(rand.Reader, length)
+		if err != nil {
+			s.logger.Error("failed to find random int", "err", err)
+			return
+		}
+
+		to, err = rand.Int(rand.Reader, length)
+		if err != nil {
+			s.logger.Error("failed to find random int", "err", err)
+			return
+		}
+
+		if from.Cmp(to) == 0 {
+			continue
+		} else {
+			return
+		}
+	}
+}
+
+func (s *StressTest) prepareDeployErc20(input *types.Block) (*types.Transaction, error) {
+	gasUsed := int64(2000_000)
+
+	maxFeePerGas := input.Transactions()[1].GasFeeCap()
+	oneHundredTen := big.NewInt(150)
+	maxFeePerGas.Mul(maxFeePerGas, oneHundredTen)
+	maxFeePerGas.Div(maxFeePerGas, big.NewInt(100))
+
+	nonceMu.Lock()
+	txn := types.NewTx(&types.DynamicFeeTx{
+		Nonce:     nonces[0],
+		Value:     big.NewInt(0),
+		Gas:       uint64(gasUsed),
+		GasTipCap: input.Transactions()[1].GasTipCap(),
+		GasFeeCap: big.NewInt(999_999_999_999),
+		Data:      common.Hex2Bytes(StressTokenBin),
+	})
+	nonceMu.Unlock()
+
+	txn.Hash()
+	signer := types.NewLondonSigner(s.cfg.ChainId)
+	signedTx, err := types.SignTx(txn, signer, privateKeys[0])
+	if err != nil {
+		s.logger.Error("failed to sign tx", "err", err)
+		return nil, err
+	}
+
+	deployTxHash = signedTx.Hash()
+	nonceMu.Lock()
+	nonces[0] += 1
+	nonceMu.Unlock()
+
+	return signedTx, nil
+}
+
+func (s *StressTest) prepareTransferErc20(input *types.Block) types.Transactions {
+	gasInfo := input.Transactions()[len(input.Transactions())-1]
+
+	originalGasUsed := input.GasUsed()
+	targetUsage := s.cfg.GasTarget
+
+	fillUpRemaining := int64(targetUsage)
+	s.logger.Info("gas used", "blockNum", input.NumberU64(), "original", originalGasUsed, "target", targetUsage)
+
+	result := types.Transactions{}
+
+	if fillUpRemaining <= 0 {
+		return result
+	}
 
-		val, ok := new(big.Int).SetString("100000000000000000000", 10)
-		if !ok {
-			panic("big.Int SetString failed")
+	for {
+		gasUsed := int64(23_000)
+		if fillUpRemaining < gasUsed {
+			break
 		}
 
-		dep.SourceHash = source.SourceHash()
-		dep.From = addr
-		dep.To = &addr
-		dep.Mint = val
-		dep.Value = val
-		dep.Gas = 21_000
-		dep.IsSystemTransaction = false
+		fillUpRemaining -= gasUsed
 
-		depositTxns = append(depositTxns, types.NewTx(&dep))
+		maxFeePerGas := gasInfo.GasFeeCap()
+		oneHundredTen := big.NewInt(150)
+		maxFeePerGas.Mul(maxFeePerGas, oneHundredTen)
+		maxFeePerGas.Div(maxFeePerGas, big.NewInt(100))
 
 		nonceMu.Lock()
-		nonces[i] += 1
+		to := common.BigToAddress(big.NewInt(int64(nonces[0])))
```

Comment:
It might be better to generate a new random address each time here. That way we should be hitting a more diverse set of merkle trie paths when reading/writing to the contract storage. That would make it a more stressful stress test, and also probably a bit more realistic

Reply:
Now I use nonce to generate the hash, making the address more random:

`erc20ToAddr := crypto.Keccak256(big.NewInt(int64(nonces[0])).Bytes()[:])[12:]`